### PR TITLE
Konflux: Add more metrics

### PR DESCRIPTION
**Pull request by danilo-gemoli**
Add the following metrics:
```
ephemeralcluster_count{konflux_cluster, konflux_namespace, cluster_profile, workflow, phase}
ephemeralcluster_provisioning_duration_seconds{workflow}
ephemeralcluster_deprovisioning_duration_seconds{workflow}
```

`ephemeralcluster_count` replace `ephemeralclusters` as the new name is much clearer and more consistent with Prometheus metric naming conventions.

## Summary

The ephemeral cluster controller now exposes Prometheus metrics for Konflux cluster operations:

- `ephemeralcluster_count` tracks clusters by profile, workflow, phase, namespace, and management cluster.
- Provisioning and deprovisioning histograms track operation duration by workflow.
- Deprovisioning metrics support restart and Prow job completion edge cases.

The reconciler records these metrics when clusters become ready or complete deprovisioning. Tests cover provisioning, failure, deprovisioning, hosted management clusters, and successful Prow job completion.

### CODE REVIEW

File: pkg/controller/ephemeralcluster/testutil_test.go [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+}
+
+func fakeScheme(t *testing.T) *runtime.Scheme {
+	scheme := runtime.NewScheme()
+	sb := runtime.NewSchemeBuilder(ephemeralclusterv1.AddToScheme, prowv1.AddToScheme, corev1.AddToScheme)
+	if err := sb.AddToScheme(scheme); err != nil {
+		t.Fatal("build scheme")
+	}
+	return scheme
+}
```
[the remark this answers is not in the document]
Comment (danilo-gemoli) [reply pkg/controller/ephemeralcluster/testutil_test.go:171]: This is minor, won't do.

File: pkg/controller/ephemeralcluster/reconciler.go [hunk 2]
```diff
@@ -1012,6 +1036,31 @@ func (r *reconciler) notifyTestComplete(ctx context.Context, log *logrus.Entry,
...
+	}
+}
+
+func (r *reconciler) collectDeprovisioningDurationMetric(ec *ephemeralclusterv1.EphemeralCluster, oldStatus, observedStatus *ephemeralclusterv1.EphemeralClusterStatus) {
+	if oldStatus.Phase != ephemeralclusterv1.EphemeralClusterDeprovisioned && observedStatus.Phase == ephemeralclusterv1.EphemeralClusterDeprovisioned {
+		if !r.collectDeprovisioningDurationMetricFunc(ec, oldStatus, observedStatus) {
+			r.logger.Warn("Failed to collect the deprovisioning duration metric")
+		}
+	}
+}
```
[the remark this answers is not in the document]
Comment (danilo-gemoli) [reply pkg/controller/ephemeralcluster/reconciler.go:1062]: Minor, won't do it.

**Review by droslean — APPROVED** [review]
